lib/maths/CBoostedTreeFactory.cc

```diff
@@ -124,6 +124,14 @@ CBoostedTreeFactory::buildFor(core::CDataFrame& frame, std::size_t dependentVari
         this->initializeHyperparameterOptimisation();
     }
 
+    m_TreeImpl->m_BestHyperparameters =
+        CBoostedTreeHyperparameters(m_TreeImpl->m_Regularization,
+                                    m_TreeImpl->m_DownsampleFactor,
+                                    m_TreeImpl->m_Eta,
+                                    m_TreeImpl->m_EtaGrowthRatePerTree,
+                                    m_TreeImpl->m_MaximumNumberTrees,
+                                    m_TreeImpl->m_FeatureBagFraction);
```

Comment:
Minor thing, can we move this into the end of `initializeHyperparameterOptimisation`. Otherwise, I need to duplicate for build for incremental training.

Reply:
Done.